core-web/libs/sdk/react/src/lib/hooks/useDotcmsEditor.ts

```diff
@@ -4,26 +4,51 @@ import {
     DotCMSPageEditorConfig,
     destroyEditor,
     initEditor,
-    isInsideEditor,
-    updateNavigation
+    isInsideEditor as isInsideEditorFn,
+    updateNavigation,
+    CUSTOMER_ACTIONS,
+    postMessageToEditor
 } from '@dotcms/client';
+
+import { DotCMSPageContext } from '../models';
 export const useDotcmsEditor = (config?: DotCMSPageEditorConfig) => {
-    const [isInsideEditorPage, setIsInsideEditorPage] = useState(false);
+    const [isInsideEditor, setIsInsideEditor] = useState(false);
+    const [pageInfo, setPageInfo] = useState<DotCMSPageContext['pageAsset'] | null>(null);
+
+    const handlePostMessage = (event: MessageEvent) => {
+        const insideEditor = isInsideEditorFn();
+        if (!insideEditor) {
+            return;
+        }
+
+        if (event.data.name === 'SET_PAGE_INFO') {
+            setPageInfo(event.data.payload);
+        }
+    };
+
     useEffect(() => {
-        const insideEditor = isInsideEditor();
+        const insideEditor = isInsideEditorFn();
         if (insideEditor) {
-            initEditor(config);
+            initEditor();
             updateNavigation(config?.pathname || '/');
+            postMessageToEditor({
+                action: CUSTOMER_ACTIONS.GET_PAGE_INFO,
+                payload: {
+                    pathname: config?.pathname
+                }
+            });
         }
 
-        setIsInsideEditorPage(insideEditor);
+        setIsInsideEditor(insideEditor);
+        window.addEventListener('message', handlePostMessage);
```

Comment:
This and the Angular listener should happen inside the `@dotcms/client` which is the library that handles all the post messages:

1. If we change the method of communication from postMessage to let's way web socket the Angular, React and any other library we write should not care about.
2. One source of true
3. Unified communication source, any data manipulation happen in one place.

Reply:
Unified!